core/src/main/java/org/aerogear/mobile/core/metrics/MetricsPublisher.java

```diff
@@ -1,12 +1,17 @@
 package org.aerogear.mobile.core.metrics;
 
+import android.support.annotation.Nullable;
+
+import org.aerogear.mobile.core.Callback;
+
 public interface MetricsPublisher {
 
     /**
      * Allows to publish metrics to external source
-     * 
+     *
      * @param metrics a array of metrics objects to publish
+     * @param callback callback of the publication
      */
-    void publish(Metrics... metrics);
+    void publish(Metrics[] metrics, @Nullable final Callback callback);
```

Comment:
add another publish method here as suggested in the other comment

Reply:
Both in `MetricsService` and `MetricsPublisher`? But no one is going to implement that method, in `MetricsService` I simply added:
```
    public void publish(Metrics... metrics) {
        publish(metrics, null);
    }
```
And `MetricsService` don't implement that interface anyway.

Reply:
You could add a `default implementation`. However, if you think it would not be useful or used at all, don't add it.